lib/pbench/server/api/resources/users_api.py

```diff
@@ -324,36 +339,26 @@ def get(self, username):
                     }
         """
         try:
-            user, verified = self.auth.verify_user(username)
-        except Exception:
-            self.logger.exception("Exception occurred during verifying the user")
-            abort(500, message="INTERNAL ERROR")
-
-        # TODO: Check if the user has the right privileges
-        if verified:
-            response_object = user.get_json()
-            return make_response(jsonify(response_object), 200)
-        elif user.is_admin():
-            try:
-                target_user = User.query(username=username)
+            target_user = self.operation_permitted(target_username)
+            if target_user:
                 response_object = target_user.get_json()
-                return make_response(jsonify(response_object), 200)
-            except Exception:
-                self.logger.exception(
-                    "Exception occurred while querying the user. Username: {}", username
+                return make_response(jsonify(response_object), HTTPStatus.OK)
+            else:
+                self.logger.warning(
+                    "User {} is not authorized to get user {}.",
+                    self.auth.token_auth.current_user().username,
+                    target_username,
                 )
-                abort(500, message="INTERNAL ERROR")
-        else:
-            self.logger.warning(
-                "User {} is not authorized to get user {}.", user.username, username,
-            )
-            abort(
-                403,
-                message=f"Not authorized to access information about user {username}",
-            )
+                abort(
+                    HTTPStatus.FORBIDDEN,
+                    message=f"Not authorized to access information about user {target_username}",
+                )
+        except Exception:
+            self.logger.exception("Exception occurred during verifying the user")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
if a logged-in user makes a GET request for a non-existent user, it is a `FORBIDDEN` since `get_valid_target_user` method would return `None` in this case because it couldn't find the target user. It's an interesting idea that an admin_user should get a `NOT_FOUND` instead of forbidden